src/veh_interact.cpp

```diff
@@ -371,6 +378,15 @@ void veh_interact::cache_tool_availability()
                 crafting_inv.has_components( "wheel_bicycle", 1 ) ||
                 crafting_inv.has_components( "wheel_motorbike", 1 ) ||
                 crafting_inv.has_components( "wheel_small", 1 );
+
+    int qual_jack = ceil( double( veh->total_mass() * 1000 ) / TOOL_LIFT_FACTOR );
+    has_jack = g->u.has_quality( "JACK", qual_jack ) ||
+               map_selector( g->u.pos(), PICKUP_RANGE ).has_quality( "JACK", qual_jack ) ||
+               vehicle_selector( g->u.pos(), 1, *veh ).has_quality( "JACK", qual_jack );
```

Comment:
> Consistent behavior in all circumstances

Inconsistent with crafting and map uses and annoying, for no good reason.

> If you're working on a sufficiently large vehicle you're already going to have to haul tools around due to PICKUP_RANGE.

Nope - you will just need to work on the tiles that are further away. Same with crane - it has the range of an entire vehicle, you just need to stand next to it and the vehicle.

> Sure, but in another scenario you might also be using your main vehicle to salvage parts from another (much larger) vehicle/wreckage.

If you're worried about safety, you can still do it with 1 tile range, by installing the crane 1 tile deep into the vehicle and a door next to it. Regardless of the placement, you still need to park pretty exactly next to the vehicle you want to operate on, which is a tile-counting kind of annoyance with no positive effects for the gameplay.